### PR TITLE
feat(logger): add trace & span info to GCP format

### PR DESCRIPTION
**Pull request by yashafromrussia**
We need to format the trace & span info according to Google Logging format - https://cloud.google.com/logging/docs/structured-logging

We expect the trace & span info to be injected via https://www.npmjs.com/package/@opentelemetry/instrumentation-winston

### CODE REVIEW

File: packages/logger/src/format/common.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { TransformableInfo } from 'logform';
```
Comment (yashafromrussia) [inline packages/logger/src/format/common.ts:1]: The contents of this file were moved from `packages/logger/src/format.ts`

File: packages/logger/src/format/constants.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const ERROR_PARAM_POSITION = 0;
```
Comment (yashafromrussia) [inline packages/logger/src/format/constants.ts:1]: The contents of the file were moved from `packages/logger/src/format.ts`

File: packages/logger/src/format/gcp.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import { format } from 'winston';
+import { getWinstonParams, isTraceSampled } from './helpers';
+import { TransformableInfo } from 'logform';
+import { ERROR_PARAM_POSITION } from './constants';
+
+/**
+ * Creates a format function that formats the log entry in a GCP friendly format - either for error or regular log entries.
+ */
+const commonFieldsFormat = format(
```
Comment (yashafromrussia) [inline packages/logger/src/format/gcp.ts:9]: This is what adds the new functionality

File: packages/logger/src/format/gcp.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      spanId: span_id,
+      traceSampled: isTraceSampled(trace_flags),
+    };
+  },
+);
+
+/**
+ * Creates a format function that formats the error log entry in a GCP friendly format.
+ */
+const errorFormat = format((info: TransformableInfo): TransformableInfo => {
```
Comment (yashafromrussia) [inline packages/logger/src/format/gcp.ts:27]: Error format has not been changed

File: packages/logger/src/format/gcp.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    label: `${message.replace(error.message, '')}`,
+    message: error.stack,
+  };
+});
+
+const gcpLogFormat = commonFieldsFormat();
+const gcpErrorFormat = format.combine(
+  gcpLogFormat,
+  errorFormat(),
+);
```
Comment (yashafromrussia) [inline packages/logger/src/format/gcp.ts:66]: Adding trace & span info to both - errors and regular logs

File: packages/logger/src/format/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import stringify from 'json-stringify-safe';
```
Comment (yashafromrussia) [inline packages/logger/src/format/helpers.ts:1]: The contents of the file were moved from `packages/logger/src/format.ts`

**Review by nikolaev-k — APPROVED** [review]

**Review by maryfranks — APPROVED** [review]

**Pull request merged**